multinode-demo/common.sh

```diff
@@ -39,17 +39,20 @@ else
       program="solana-$program"
     fi
 
-    if [[ -r "$SOLANA_ROOT/$crate"/Cargo.toml ]]; then
-      maybe_package="--package solana-$crate"
-    fi
     if [[ -n $NDEBUG ]]; then
       maybe_release=--release
     fi
-    declare manifest_path="--manifest-path=$SOLANA_ROOT/$crate/Cargo.toml"
-    printf "cargo $CARGO_TOOLCHAIN run $manifest_path $maybe_release $maybe_package --bin %s %s -- " "$program"
+
+    (
+      set -x
+      # Prebuild local binaries so that CI sanity check doesn't include build time
+      # in the check timeout
+      cargo $CARGO_TOOLCHAIN build $maybe_release --bin $program
```

Comment:
By excluding the manifest path and package arguments, builds are properly cached. Need to verify if this breaks CUDA builds though

Reply:
CUDA builds look fine: https://buildkite.com/solana-labs/solana/builds/44119#78520d88-a3a7-41c8-82e4-e394fb8a1ba6